File: project-java-BogdanBala112/HomeAssignment1/src/main/java/ControllerEntities/TicketController.java
package ControllerEntities;

import Entities.Ticket;
import Repository.TicketRepository;
import javafx.collections.FXCollections;
import javafx.collections.ObservableList;
import service.TicketService;

import java.util.List;

public class TicketController {
    private final TicketService ticketService;
    private final ObservableList<Ticket> ticketList;

    public TicketController() {
        TicketRepository repository = new TicketRepository();
        this.ticketService = new TicketService(repository);
        this.ticketList = FXCollections.observableArrayList(ticketService.getTickets());
    }

    public ObservableList<Ticket> getTicketList() {
        return ticketList;
    }

    public void addTicket(Ticket ticket) {
        int size = ticketService.getTickets().size();
        ticketService.addTicket(ticket);
        if (size + 1 == ticketService.getTickets().size())
            ticketList.add(ticket);
    }

    public void deleteTicket(int id) {
        ticketService.deleteTicket(id);
        ticketList.removeIf(m -> m.getticketId() == id);
    }

    public void updateTicket(int id, Ticket ticket) {
        ticketService.updateTicket(id, ticket);
        ticketList.clear();
        ticketList.addAll(ticketService.getTickets());
    }

    public void loadTicketsFromDatabase() {
        List<Ticket> freshTickets = ticketService.getTickets();
        ticketList.setAll(freshTickets);
    }
}

File: ControllerEntities/TicketController.cs
using System.Collections.ObjectModel;
using System.Linq;
using Domain;
using projectC.Repository;
using projectC.Service;

namespace projectC.ControllerEntities
{
    public class TicketController
    {
        private readonly TicketService ticketService;
        public ObservableCollection<Ticket> TicketList { get; }

        public TicketController()
        {
            var repository = new TicketRepository();
            ticketService = new TicketService(repository);
            TicketList = new ObservableCollection<Ticket>(ticketService.GetTickets());
        }

        public void AddTicket(Ticket ticket)
        {
            ticketService.AddTicket(ticket);
            TicketList.Add(ticket);
        }

        public void DeleteTicket(int id)
        {
            ticketService.DeleteTicket(id);
            var ticketToRemove = TicketList.FirstOrDefault(t => t.ticketId == id);
            if (ticketToRemove != null) TicketList.Remove(ticketToRemove);
        }

        public void UpdateTicket(int id, Ticket ticket)
        {
            ticketService.UpdateTicket(id, ticket);
            RefreshTickets();
        }

        public void RefreshTickets()
        {
            TicketList.Clear();
            foreach (var t in ticketService.GetTickets())
                TicketList.Add(t);
        }
    }
}